### ISSUE TITLE
[I386] llvm-toolchain 15: 32-bit specific test failures

### ISSUE THREAD

**Issue by kasperk81** [llvm/llvm-project#57401]
llvm 15 rc2 build fail when running tests https://buildd.debian.org/status/fetch.php?pkg=llvm-toolchain-15&arch=i386&ver=1%3A15.0.0%7E%2Brc2-1%7Eexp1&stamp=1660118537&raw=0

excerpt:
```console
[572/573] Linking CXX executable unittests/tools/llvm-profgen/LLVMProfgenTests
[572/573] Running the LLVM regression tests
-- Testing: 42732 tests, 4 workers --
Testing: 
FAIL: LLVM :: Analysis/CostModel/ARM/add-cast-vect.ll (308 of 42732)
******************** TEST 'LLVM :: Analysis/CostModel/ARM/add-cast-vect.ll' FAILED ********************
Script:
--
: 'RUN: at line 1';   /<<PKGBUILDDIR>>/build-llvm/tools/clang/stage2-bins/bin/opt < /<<PKGBUILDDIR>>/llvm/test/Analysis/CostModel/ARM/add-cast-vect.ll  -passes='print<cost-model>' 2>&1 -disable-output -mtriple=armv7-linux-gnueabihf -mcpu=cortex-a9 | /<<PKGBUILDDIR>>/build-llvm/tools/clang/stage2-bins/bin/FileCheck --check-prefix=COST /<<PKGBUILDDIR>>/llvm/test/Analysis/CostModel/ARM/add-cast-vect.ll
--
Exit Code: 1

Command Output (stderr):
--
/<<PKGBUILDDIR>>/llvm/test/Analysis/CostModel/ARM/add-cast-vect.ll:24:9: error: COST: expected string not found in input
; COST: cost of 1 for instruction: {{.*}} add <4 x i32>
        ^
<stdin>:1:62: note: scanning from here
Printing analysis 'Cost Model Analysis' for function 'direct':
                                                             ^
<stdin>:4:33: note: possible intended match here
Cost Model: Found an estimated cost of 4 for instruction: %r3 = add <4 x i32> %v0, %v1
                                ^

Input file: <stdin>
Check file: /<<PKGBUILDDIR>>/llvm/test/Analysis/CostModel/ARM/add-cast-vect.ll

-dump-input=help explains the following input dump.
```

rc3 build is overshadowed by another issue https://github.com/llvm/llvm-project/issues/57389#issuecomment-1229202744, so we don't get to the tests.

**Comment by kasperk81** [comment]
this is i386 machine not arm (costmodel test is one of thousands of tests which are failing which happen to be in /ARM directory)

**Comment by EugeneZelenko** [comment]
@kasperk81: but aren't failed tests related to ARM?

**Comment by kasperk81** [comment]
no, those are analyzer test `/llvm/test/Analysis/CostModel/ARM/add-cast-vect.ll` file is just a fixture in the context of test.
see the log link posted above, there are many other failing tests due to "expected string not found in input", i just copied the first failing test. here is another example:
```
/<<PKGBUILDDIR>>/clang/test/CodeGen/address-safety-attr.cpp:84:12: error: BLFILE: expected string not found in input
// BLFILE: AddressSafetyOk{{.*}}) [[NOATTR]]
           ^
<stdin>:163:62: note: scanning from here
define noundef i32 @_Z16NoAddressSafety6Pi(ptr noundef %a) #3 {
                                                             ^
<stdin>:163:62: note: with "NOATTR" equal to "#3"
define noundef i32 @_Z16NoAddressSafety6Pi(ptr noundef %a) #3 {
                                                             ^
<stdin>:173:25: note: possible intended match here
define noundef i32 @_Z15AddressSafetyOkPi(ptr noundef %a) #0 {
                        ^

Input file: <stdin>
Check file: /<<PKGBUILDDIR>>/clang/test/CodeGen/address-safety-attr.cpp

-dump-input=help explains the following input dump.
```

**Comment by kasperk81** [comment]
```
-- Could NOT find Python module yaml
-- LLVM host triple: i386-linux-gnu
-- LLVM default target triple: i386-linux-gnu
-- Performing Test CXX_SUPPORTS_CUSTOM_LINKER
```

**Comment by tru** [comment]
This seems like a single test failure which I don't think is important enough to be fixed before 15.0.0-final. I will move this to 15.0.1 and hope someone will have a look by then.

**Comment by kasperk81** [comment]
new build fixed the issue https://buildd.debian.org/status/fetch.php?pkg=llvm-toolchain-15&arch=i386&ver=1%3A15.0.0%7E%2Brc3-1%7Eexp2&stamp=1661995996&raw=0
